File: old-app-java/src/main/java/tech/thdev/java_udemy_sample/data/FlickrOwner.java
package tech.thdev.java_udemy_sample.data;

/**
 * Created by tae-hwan on 11/17/16.
 */

public class FlickrOwner {

    public String nsid;
    public String username;
    public String realname;
    public String location;
    public String iconserver;
    public int iconfarm;
    public String path_alias;

    public String getBuddyiconUrl() {
        return String.format("https://farm%s.staticflickr.com/%s/buddyicons/%s.jpg", iconfarm, iconserver, nsid);
    }
}

File: old-app-kotlin/src/main/java/tech/thdev/kotlin_udemy_sample/data/FlickrOwner.kt
package tech.thdev.kotlin_udemy_sample.data

/**
 * Created by tae-hwan on 11/13/16.
 */

data class FlickrOwner(val nsid: String,
                       val username: String,
                       val realname: String,
                       val location: String,
                       val iconserver: String,
                       val iconfarm: Int,
                       val path_alias: String) {

    /**
     * Buddyicons API : <a href="https://www.flickr.com/services/api/misc.buddyicons.html">api</a>
     * http://farm{icon-farm}.staticflickr.com/{icon-server}/buddyicons/{nsid}.jpg
     */
    fun getBuddyIcons()
        = "http://farm$iconfarm.staticflickr.com/$iconserver/buddyicons/$nsid.jpg"
}